### ISSUE TITLE
Global private variables cause `reportPrivateUsage` error when used in classes

### ISSUE THREAD

**Issue by max-muoto** [microsoft/pyright#7884]
Normally, it's more than fine to assign a global private variable in the same module.

```python
_my_private_var: int | None = None

def foo() -> None:
    # No errors.
    global _my_private_var
    _my_private_var = 3
```

However, introducing usage of said variable into a class makes Pyright think it's scoped to that class, and will report all declarations outside as being a violation of that attribute being private.

```python
_my_private_var: int | None = None

def foo() -> None:
    global _my_private_var
    _my_private_var = 3

class MyClass:
    def foo(self) -> None:
        # All usages outside of this method are flagged.
        global _my_private_var
        _my_private_var = 3
```

[Pyright Playground Example
](https://pyright-play.net/?pyrightVersion=1.1.358&pythonVersion=3.12&strict=true&code=PoWwnsAOBOCWBuBDALgU2E6AuABLAdsjgD44ByA9vqjgLzlWoBQLTAJqgGY6cUUAUAShwBaAHwNqWJjlk4A5gBsKAI0SKcoCDAQp0mGXK1Q4SNBkTQ6OAMwsAxosQBnZzgCyYAMJPX0uTgc3LwCzqiKnMLikqj%2BAXJKquqa4Ca65gbxRqk6ZvqW1nZAA)

**VS Code extension or command-line**
1.1.362, both VSCode and the CLI tool.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.363, which I just published.
